### PR TITLE
Add hugging face GPT2 model.

### PR DESCRIPTION
**Pull request by caiom**
1. Create an abstract model, ArchaiModel, from which all model must derive to be compatible with the codebase (train.py)
2. Adapts huggingface GPT2 model to our codebase (hf_gpt2.py)
3. Move all models to the 'models' folder.

### CODE REVIEW

File: archai/nlp/nvidia_transformer_xl/models/archai_model.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from abc import ABC, abstractmethod
```
Comment (gugarosa) [inline archai/nlp/nvidia_transformer_xl/models/archai_model.py:1]: Could you please remove `ABC` as it is not used?

File: archai/nlp/nvidia_transformer_xl/models/archai_model.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            else torch.device('cpu')
+
+        # Loads the checkpoint
+        checkpoint = torch.load(path, map_location=dst)
+
+        model_config = checkpoint['model_config']
+
+        # Compatibility with older models
+        # TODO: remove once not needed
+        if 'encoder_like' in model_config:
```
Comment (gugarosa) [inline archai/nlp/nvidia_transformer_xl/models/archai_model.py:45]: Do we still need this?

**Comment by gugarosa** [comment]
Apart from the comments, looks awesome to me!

File: archai/nlp/nvidia_transformer_xl/models/archai_model.py (hunk 2 above)
Comment (caiom) [reply archai/nlp/nvidia_transformer_xl/models/archai_model.py:45]: Not really, will remove

**Pull request merged**